cstore_reader.c

```diff
@@ -1144,7 +1176,62 @@ DeserializeBlockData(StripeBuffers *stripeBuffers, uint64 blockIndex,
 			/* store current block's data buffer to be freed at next block read */
 			blockData->valueBuffer = valueBuffer;
 		}
+		else if (columnAdded)
+		{
+			/* populate exists and value buffers for missing column */
+			int rowIndex = 0;
+			bool useDefaultValue = attributeForm->atthasdef;
+			Datum defaultValue = (Datum) NULL;
+
+			if (useDefaultValue)
+			{
+				useDefaultValue = DefaultValueForColumn(tupleDescriptor->constr,
+													 columnIndex, &defaultValue);
+			}
+
+			for (rowIndex = 0; rowIndex < rowCount; rowIndex++)
+			{
+				blockData->existsArray[rowIndex] = useDefaultValue;
+				/* it is safe to assign datum value since this array content
+				 * is never deallocated.
+				 */
+				blockData->valueArray[rowIndex] = defaultValue;
+			}
+		}
+	}
+}
+
+
+/*
+ * DefaultValueForColumn returns default value for given column.
+ * Only const values are supported. The function returns NULL for all other
+ * default value types.
```

Comment:
I am more into checking this during alter command as we discussed. Per your suggestion I will check immutable functions too